models/correlospinmatrix_detector.h

```diff
@@ -155,11 +155,18 @@ class correlospinmatrix_detector : public Node
    */
   using Node::handle;
   using Node::handles_test_event;
+  using Node::receives_signal;
 
   void handle( SpikeEvent& );
 
   port handles_test_event( SpikeEvent&, rport );
 
+  SignalType
+  receives_signal() const
+  {
+    return BINARY;
+  }
+
```

Comment:
Please move the function definition out of the class definition. Inline functions should be defined towards the end of the header file.